ape_solidity/compiler.py

```diff
@@ -738,6 +729,50 @@ def enrich_error(self, err: ContractLogicError) -> ContractLogicError:
                 txn=err.txn,
             )
 
+    def _flatten_source(self, path: Path, base_path=None) -> str:
+        base = base_path or path.parent
+        imports = self.get_imports_with_raw([path])
+        source = ""
+        for import_list in imports.values():
+            if not import_list:
+                continue
+            for import_path, raw_import_path in import_list:
+                source += f"// File: {raw_import_path}\n\n"
+                source += self._flatten_source(base / import_path, base_path=base)
+        if base_path is None:
+            source += f"\n// File: {path.name}\n\n"
+        source += path.read_text() + "\n"
+        return source
+
+    def flatten_contract(self, path: Path, **kwargs) -> Content:
+        """Flatten a contract.
```

Comment:
doc str is not super needed here (it will API base class if not given one)
unless there is some unique Solidity-isms you wish to call out.